dyninstAPI_RT/src/RTheap.c

```diff
@@ -279,75 +276,31 @@ void *DYNINSTos_malloc(size_t nbytes, void *lo_addr, void *hi_addr)
     if (ret_heap < (Address)lo_addr ||
         ret_heap + size - 1 > (Address)hi_addr) {
       free(heap);
-      free(node);
 #ifdef DEBUG
       fprintf(stderr, "MALLOC'd area fails range constraints\n");
 #endif 
       return NULL;
     }
 
     /* define new heap */
+    node = heap + size;
```

Comment:
I think it should be `node = ret_heap + size`, but in practice `malloc` probably aligns sufficiently anyway.